python/graphstorm/gconstruct/construct_graph.py

```diff
@@ -336,37 +356,74 @@ def process_labels(data, label_confs):
         The list of configs to construct labels.
+    is_node : bool
+        Whether or not to process labels on nodes.
+
+    Returns
+    -------
+    dict of tensors : labels (optional) and train/validation/test masks.
     """
     assert len(label_confs) == 1, "We only support one label per node/edge type."
     label_conf = label_confs[0]
-    assert 'label_col' in label_conf, "'label_col' must be defined in the label field."
-    label_col = label_conf['label_col']
-    label = data[label_conf['label_col']]
     assert 'task_type' in label_conf, "'task_type' must be defined in the label field."
     if label_conf['task_type'] == 'classification':
-        assert np.issubdtype(label.dtype, np.integer), \
+        assert 'label_col' in label_conf, \
+                "'label_col' must be defined in the label field."
+        label_col = label_conf['label_col']
+        label = data[label_col]
+        assert np.issubdtype(label.dtype, np.integer) \
+                or np.issubdtype(label.dtype, np.floating), \
                 "The labels for classification have to be integers."
+        valid_label_idx = get_valid_label_index(label)
         label = np.int32(label)
+        num_samples = len(label)
+    elif label_conf['task_type'] == 'regression':
+        assert 'label_col' in label_conf, \
+                "'label_col' must be defined in the label field."
+        label_col = label_conf['label_col']
+        label = data[label_col]
+        valid_label_idx = get_valid_label_index(label)
+        num_samples = len(label)
+    else:
+        assert label_conf['task_type'] == 'link_prediction', \
+                "The task type must be classification, regression or link_prediction."
+        assert not is_node, "link_prediction task must be defined on edges."
+        label_col = label = None
+        valid_label_idx = None
+        # Any column in the data can define the number of samples in the data.
+        for val in data.values():
+            num_samples = len(val)
+            break
+
     if 'split_pct' in label_conf:
         train_split, val_split, test_split = label_conf['split_pct']
         assert train_split + val_split + test_split <= 1, \
                 "The data split of training/val/test cannot be more than the entire dataset."
-        rand_idx = np.random.permutation(len(label))
-        train_idx = rand_idx[0:int(len(label) * train_split)]
-        val_start = int(len(label) * train_split)
-        val_end = int(len(label) * (train_split + val_split))
+        if valid_label_idx is None:
+            rand_idx = np.random.permutation(num_samples)
+        else:
+            rand_idx = np.random.permutation(valid_label_idx)
+        num_labels = len(rand_idx)
+        train_idx = rand_idx[0:int(num_labels * train_split)]
+        val_start = int(num_labels * train_split)
+        val_end = int(num_labels * (train_split + val_split))
         val_idx = rand_idx[val_start:val_end]
         test_idx = rand_idx[val_end:]
```

Comment:
So here test_split does not take effect. We assume the rest will all become test set?

Reply:
this is to split the data with valid labels. What should we do with the rest of labelled samples?

Reply:
I mean if users give [0.5, 0.1, 0.1]. It means 50% for train, 10% for val, and 10% for test. There will be 30% unused. Base on the current logic. The test set will be 40%, which is not users want.

Reply:
i see. i'll fix it.